packages/ethers-cf-purelocal/index.ts

```diff
@@ -0,0 +1,75 @@
+import { ethers } from "ethers";
+import VM from "ethereumjs-vm";
+
+const vm = new VM({
+  enableHomestead: true,
+  activatePrecompiles: true
+});
+export class PureEvmProvider extends ethers.providers.JsonRpcProvider {
+  contracts;
+  constructor(...args) {
+    super(...args);
+    this.contracts = require(`@counterfactual/contracts/networks/${

```

Comment:
Ought to remove the dependence on our contracts.

Reply:
@snario Without including the contracts package we don't actually have the code we want to run on EVM; I will add a call to getCode() to the contract constructor, unless you have a better idea for how to implement?

p.s. I recognize the face; I think we met briefly at ETHSingapore :)

Reply:
The `Contract` object has `this.address` and `this.abi` parameters. We could fetch the bytecode from the chain. Alternatively we could use the `ContractFactory` object to pass in the bytecode but no address.

Reply:
Hm. This is where I'd love to have @ricmoo chime in...

Reply:
@snario One solution I can see here is too add the methods under their own object, e.g. Contract.localFunctions._functionName_ . Would this be acceptable for your use case?